worker-store-in-memory/src/main/java/com/hpe/caf/worker/datastore/inmemory/InMemoryDataStore.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2015-2017 Hewlett Packard Enterprise Development LP.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.worker.datastore.inmemory;
+
+import com.hpe.caf.api.HealthResult;
+import com.hpe.caf.api.worker.DataStoreException;
+import com.hpe.caf.api.worker.DataStoreMetricsReporter;
+import com.hpe.caf.api.worker.ManagedDataStore;
+import org.apache.commons.io.IOUtils;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+/**
+ * ManagedDataStore implementation for an In Memory Datastore which uses a HashMap.
+ */
+public class InMemoryDataStore implements ManagedDataStore
+{
+
+    public InMemoryDataStore()
+    {
+        // No need to do anything in constructor
+    }
+
+    /**
+     * We will use a HashMap as the data storage structure.
+     */
+    Map<String, byte[]> dataMap = new HashMap<>();
+
+    /**
+     * The Storage Service "file type" for Worker assets.
+     */
+    private final AtomicInteger errors = new AtomicInteger(0);
+    private final AtomicInteger numRetrieveRequests = new AtomicInteger(0);
+    private final AtomicInteger numStoreRequests = new AtomicInteger(0);
+    private final AtomicInteger numDeleteRequests = new AtomicInteger(0);
+    private final DataStoreMetricsReporter metrics = new InMemoryMetricsReporter();
+
+    /**
+     * Remove the asset identified by the reference.
+     * @param reference a complete reference to be interpreted by the DataStore implementation
+     * @throws DataStoreException if data store cannot service the request.
+     */
+    @Override
+    public void delete(String reference) throws DataStoreException
+    {
+        numDeleteRequests.incrementAndGet();
+        dataMap.remove(reference);
```

Comment:
If the reference does not exist, is it OK here to do this or do we want to throw an exception saying "Asset does not exist"?